### PR TITLE
Fix #8: Add the possibility to highlight specific lines

### LINKED ISSUE #8: Add the possibility to highlight specific lines
To just like GitHub does.
- We can add a parameter to a paste URL (`/paste/foo123&ln=1-2+42` for instance)
- We can pass the `hl_lines` flag to Pygments when coloring the paste text

### PR DESCRIPTION
**Pull request by MrChoclate**
We can now do /paste/foo123&ln=1-2+42.
+ is a sep
- specify range

Fix #8

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -33,16 +34,33 @@
...
         highlight(paste.content, lexer, formatter),
         formatter.get_style_defs('.source')
     )
 
 
+def parse_hl(hl):
+    if hl is None:
+        return []
+
+    if not re.match(r"\d+|(\d+-\d+)( \d+|(\d+-\d+))*", hl):
```
Comment (Thooms) [inline main.py:60]: Can you compile the regex once in the file ? Cf https://docs.python.org/3/library/re.html#re.compile

File: main.py [hunk 2]
```diff
@@ -33,16 +34,33 @@
...
     lexer = get_lexer_by_name(paste.hl_alias)
-    formatter = HtmlFormatter(linenos=True, cssclass='source')
+    formatter = HtmlFormatter(linenos=True, cssclass='source', hl_lines=hl_lines)
     return (
         highlight(paste.content, lexer, formatter),
         formatter.get_style_defs('.source')
     )
 
 
+def parse_hl(hl):
```
Comment (Thooms) [inline main.py:47]: Can you add a small comment to explain the DSL and how the code works please?

File: main.py [hunk 3]
```diff
@@ -69,9 +87,10 @@ def view_paste(paste_name=''):
     if not paste_name:
         redirect(url_for('home'))
 
+    hl_lines = parse_hl(request.args.get('ln'))
```
Comment (Thooms) [inline main.py:99]: `hl_lines_query` instead?

**Comment by Thooms** [comment]
Small comments, but otherwise LGTM, I'll merge as soon as I can test it locally :) Thanks for your work !

File: main.py (hunk 1 above)
Comment (MrChoclate) [inline main.py:60]: Well:

``` python
>>> timeit(lambda: re.match(r"\d+|(\d+-\d+)( \d+|(\d+-\d+))*", "52 2-6")) / 1000000
1.517474457003118e-06
>>> timeit(lambda: regexp.match("52 2-6")) / 1000000
8.780031549977139e-07
```

I won't bother myself for 1ns/request

**Comment by Thooms** [comment]
Awesome work, thank you!

**Pull request merged**